File: csharp/ReceiptItem.cs
namespace supermarket
{
    public class ReceiptItem
    {
        private Product product;
        private double price;
        private double totalPrice;
        private double quantity;

        public ReceiptItem(Product p, double quantity, double price, double totalPrice)
        {
            this.product = p;
            this.quantity = quantity;
            this.price = price;
            this.totalPrice = totalPrice;
        }

        public double getPrice()
        {
            return this.price;
        }

        public Product getProduct()
        {
            return product;
        }

        public double getQuantity()
        {
            return quantity;
        }

        public double getTotalPrice()
        {
            return totalPrice;
        }

    }

}

File: cpp/src/ReceiptItem.cpp
#include "ReceiptItem.h"

ReceiptItem::ReceiptItem(const Product& product, double quantity, double price, double totalPrice)
    : product(product), price(price), totalPrice(totalPrice), quantity(quantity) {}

Product ReceiptItem::getProduct() const {
    return product;
}

double ReceiptItem::getPrice() const {
    return price;
}

double ReceiptItem::getTotalPrice() const {
    return totalPrice;
}

double ReceiptItem::getQuantity() const {
    return quantity;
}

bool ReceiptItem::operator==(const ReceiptItem& rhs) const {
    return product == rhs.product &&
           price == rhs.price &&
           totalPrice == rhs.totalPrice &&
           quantity == rhs.quantity;
}

bool ReceiptItem::operator!=(const ReceiptItem& rhs) const {
    return !(rhs == *this);
}
